lightning/src/ln/channel.rs

```diff
@@ -2022,15 +2007,12 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 			}
 		}
 
-		if msg.htlc_signatures.len() != commitment_tx.1 {
-			return Err((None, ChannelError::Close(format!("Got wrong number of HTLC signatures ({}) from remote. It must be {}", msg.htlc_signatures.len(), commitment_tx.1))));
+		if msg.htlc_signatures.len() != num_htlcs {
+			return Err((None, ChannelError::Close(format!("Got wrong number of HTLC signatures ({}) from remote. It must be {}", msg.htlc_signatures.len(), num_htlcs))));
 		}
 
-		// TODO: Merge these two, sadly they are currently both required to be passed separately to
```

Comment:
The real meaning of the todo isnt solved - we still pass the HTLC set to the ChannelMonitor twice, just now once via `htlc_outputs` and once via the `HolderCommitmentTransaction`.

Reply:
ah, yes.  rewording